### ISSUE TITLE
Synchronizing Error API HTTP 404 

### ISSUE THREAD

**Issue by leolage** [LedgerHQ/satstack#96]
Hello, After 6 days waiting for initian bitoin core full syncronize, and  lot of errors on descriptors, I finally can done a lss.exe finished the start sucessfull.

When I open Ledger Live, the system try to synchorizing, but after long minutes trying, I received a error : 
Synchronizing Error API HTTP 404 

I dont see errors on lss.exe screen, what is possible a cause of this error?

I'm using Windows 10
Bitcoin Core 24

![image](https://user-images.githubusercontent.com/23029635/209532577-46a55036-ecb9-4c24-ab38-1c982dd6e726.png)

**Comment by leolage** [comment]
More details about it... in lss logs I see this 404 error, but I not found more details about this reason... after the initial sync, I can't update my accounts more.

![image](https://user-images.githubusercontent.com/23029635/212899487-9b79c07b-e6d0-461a-886e-3baa21fa3c41.png)

**Comment by adrienlacombe-ledger** [comment]
hi @leolage you still see correct balances right?

**Comment by leolage** [comment]
> 

Yes I see all transactions and balance, but not show my last purchase (tomorrow, after the initial sync)

**Comment by adrienlacombe-ledger** [comment]
what do you mean by "before sync" exactly?

**Comment by leolage** [comment]
After I configured lss and run first time, all descriptors imported first time sucessfully. After this, all balance was ok and running normaly. Later, the error starting after I generate a new address to receive bitcoin, I sent bitcoin to this address but not updated in my balance/transactions, and receive 404 error when try to sync my Ledger Live. 

I suspicious of this -1 block of LedgerLive try to scan

 sorry for my english.

**Comment by adrienlacombe-ledger** [comment]
no worries, not my mother tongue either.

hum, can you try to restart your node, just in case?

**Comment by leolage** [comment]
I already try to restart my node. I tried again import drescriptors again to check, but dont have success:
[2023/01/17 - 10:25:32]  INFO worker: Importing descriptors                         duration=30m37s progress=99.75%

[2023/01/17 - 10:25:36] ERROR ImportDescriptors - Failed to import descriptor || -8: new range must include current range = [0,1002] NumofDescriptors=40

[2023/01/17 - 10:25:36] ERROR worker: Failed while importing descriptors            error=ImportDescriptors - importdescriptor RPC failed

[2023/01/17 - 10:25:36]  INFO Shutdown server: in progress                 

[2023/01/17 - 10:25:36]  INFO AbortRescan: Abort rescan successful: false               

[2023/01/17 - 10:25:37]  INFO Unloaded wallet successfully                  wallet=satstack

**Comment by adrienlacombe-ledger** [comment]
thank you, can you try with latest version of satstack please? we released 0.18.0 recently.

**Comment by adrienlacombe-ledger** [comment]
ok, what are stastack logs when you start it normally?

**Comment by leolage** [comment]
logs when satstack start:

user@pc:~/Desktop$ ./lss 
[2023/01/17 - 11:04:09]  INFO Ledger SatStack (lss) v0.18.0                 arch=linux amd64 build= commit= runtime=go1.17.13
[2023/01/17 - 11:04:09]  INFO Config file detected                          path=/home/user/.config/Ledger Live/lss.json
[2023/01/17 - 11:04:09]  INFO Warming up...                                
[2023/01/17 - 11:04:15]  INFO Loaded existing wallet                        wallet=satstack
[2023/01/17 - 11:04:15]  INFO RPC connection established                    blockFilter=true chain=main pruned=false txindex=true
 __________________________________________
/ When Satoshi announced the first release \
| of the software, I grabbed it right      |
| away. I think I was the first person     |
| besides Satoshi to run bitcoin.          |
|                                          |
\         -- Hal Finney, 2013              /
 ------------------------------------------
        \   ^__^
         \  (₿₿)\_______
            (__)\       )\/\
                ||----w |
                ||     ||
[GIN-debug] [WARNING] Creating an Engine instance with the Logger and Recovery middleware already attached.

[GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
 - using env:	export GIN_MODE=release
 - using code:	gin.SetMode(gin.ReleaseMode)

[GIN-debug] GET    /timestamp                --> github.com/ledgerhq/satstack/httpd/handlers.GetTimestamp.func1 (3 handlers)
[GIN-debug] GET    /control/descriptors/import --> github.com/ledgerhq/satstack/httpd/handlers.ImportAccounts.func1 (3 handlers)
[GIN-debug] POST   /control/descriptors/has  --> github.com/ledgerhq/satstack/httpd/handlers.HasDescriptor.func1 (3 handlers)
[GIN-debug] GET    /blockchain/:version/explorer/_health --> github.com/ledgerhq/satstack/httpd/handlers.GetHealth.func1 (3 handlers)
[GIN-debug] GET    /blockchain/:version/explorer/status --> github.com/ledgerhq/satstack/httpd/handlers.GetStatus.func1 (3 handlers)
[GIN-debug] GET    /blockchain/:version/btc/network --> github.com/ledgerhq/satstack/httpd/handlers.GetNetwork.func1 (3 handlers)
[GIN-debug] GET    /blockchain/:version/btc/fees --> github.com/ledgerhq/satstack/httpd/handlers.GetFees.func1 (3 handlers)
[GIN-debug] GET    /blockchain/:version/btc/blocks/:block --> github.com/ledgerhq/satstack/httpd/handlers.GetBlock.func1 (3 handlers)
[GIN-debug] GET    /blockchain/:version/btc/transactions/:hash/hex --> github.com/ledgerhq/satstack/httpd/handlers.GetTransactionHex.func1 (3 handlers)
[GIN-debug] POST   /blockchain/:version/btc/transactions/send --> github.com/ledgerhq/satstack/httpd/handlers.SendTransaction.func1 (3 handlers)
[GIN-debug] GET    /blockchain/:version/btc/addresses/:addresses/transactions --> github.com/ledgerhq/satstack/httpd/handlers.GetAddresses.func1 (3 handlers)
[2023/01/17 - 11:04:15]  INFO worker: Initial Block Download complete               blockHash=000000000000000000041623bc75a198f2ec962465ad462b21528e0b0df67782 blockHeight=772389
[2023/01/17 - 11:04:15]  INFO Rescan Config file detected                   path=/home/user/.config/Ledger Live/lss_rescan.json
[2023/01/17 - 11:04:15]  INFO RescanWallet: Rescanning Wallet start_height: 772389, end_height 772389
[2023/01/17 - 11:04:15]  INFO RescanWallet: Rescan wallet was successful:  start_height: 772389, stop_height: 772389
[2023/01/17 - 11:04:15]  INFO RescanConfigFile successfully saved           path=/home/user/.config/Ledger Live/lss_rescan.json
[2023/01/17 - 11:04:22]  INFO worker: Shutdown worker: done

**Comment by adrienlacombe-ledger** [comment]
ok so satstack seems fine, thank you, I will see what I can do

**Comment by leolage** [comment]
Thanks for help, last information. I have 2 Bitcoin Wallets in my Ledger, only the second wallet, of I used yesterday to buy bitcoins have errors:

![image](https://user-images.githubusercontent.com/23029635/212982461-e8357ef8-8680-4470-8bef-7ee410953bed.png)

**Comment by adrienlacombe-ledger** [comment]
oh interesting

**Comment by pepo2121** [comment]
Hi,

today I have the same issue. 
Any news ?

![screenSho404](https://user-images.githubusercontent.com/125806550/219951388-60a9944a-0cd4-43e6-950e-e7223e0d4fd7.PNG)

**Comment by adrienlacombe-ledger** [comment]
hi @pepo2121 did you also add accounts recently?

**Comment by pepo2121** [comment]
Hi,

no, the account was added some month ago and it works fine so far.

Yesterday:
Following actions were done without success:
-->Retsart my full node
-->Update of Ledger Live
-->Clearing the cache
-->Removing my btc account
-->Try to add the account leads to never ending synchronizing (Progress bar within ledger live)

Ok, last but not least I decided to disconnect the full node.
-->Clear cache
-->restart of ledger live
-->Add the account,- I was able to select my previous account
-->Add my full node
-->Scan/Synch works fine


Today I  started Satstack and ledger live:
Same issue as before, damn :-(

**Comment by adrienlacombe-ledger** [comment]
@pepo2121 latest Ledger Live is no longer working with satstack, please stay on 2.52.

**Comment by oullec** [comment]
@adrienlacombe-ledger : 2.52 is compatible.... Thanks for letting us know, I tried to understand for a while...
So 2.53 not compatible as well ?
Thanks

**Comment by adrienlacombe-ledger** [comment]
correct @oullec

**Comment by avatar4d** [comment]
This is still not fixed for 2.57.0?

**Comment by adrienlacombe-ledger** [comment]
> This is still not fixed for 2.57.0?

no, please stay on 2.52 for now.

**Comment by NeNo87** [comment]
Hi, still issues on new ledger live versions as 2.60? Do we have to stay on 2.52?

**Comment by adrienlacombe-ledger** [comment]
> Hi, still issues on new ledger live versions as 2.60? Do we have to stay on 2.52?

Yup

**Comment by NeNo87** [comment]
> > Hi, still issues on new ledger live versions as 2.60? Do we have to stay on 2.52?
> 
> Yup

ok thanks. But i'm still facing issues when routing satstack with tor, sometimes even if "lss_rescan.json" is present satstack rescan the complete wallet as the first time.

**Comment by d-rez** [comment]
Dear Ledger, how come has this not been fixed in 7 months?

satstack v0.19
Live v2.64.2

I finally executed `lss` on my raspberry pi to let it import descriptors (if you interrupt / pause this process, it will just restart from zero, and I have no intention of keeping my power-hungry PC running for 24h), copied over config files and getting tens of `404` errors and synchronisation failed.

At this point I haven't been able to use the full node feature properly for well over a year.

**Comment by adrienlacombe-ledger** [comment]
> Dear Ledger, how come has this not been fixed in 7 months?
> 
> satstack v0.19 Live v2.64.2
> 
> I finally executed `lss` on my raspberry pi to let it import descriptors (if you interrupt / pause this process, it will just restart from zero and then crash at some point, requiring the satstack wallet file to be deleted and recreated), and I have no intention of keeping my power-hungry PC running for 24h),
> 
> Copied over config files and getting tens of `404` errors and synchronisation failed. At this point I haven't been able to use the full node feature properly for well over a year.

please stay on LEeger Live Desktop 2.52.0

**Comment by d-rez** [comment]
Thanks for your reply.

> satstack runs on the same machine as Ledger Live, not the machine running the node.

I am aware, but this is not my scenario either. I run the node on machine A, executed `lss` descriptor sync on machine B and once it was done copied the sync status file to machine C where Ledger Live lives so that I wouldn't have to keep machine C running for prolonged periods of time.

This satstack logic could use some improvement, it is illogical to expect every user to be forced into keeping their PC on 24/7 until descriptors are imported. Based on my experience so far, any and all interruptions in this process will need the whole a full rescan. Some PCs pull 200-300W just being idle. But I digress.

> Please stay on Ledger Live Desktop 2.52.0

Thanks for the suggestion, I'll look into downgrading and decide if this is an option I am willing to pursue.

**Comment by MartinacPlus** [comment]
Has there been any progress on this since last December?

Because I too get the "Synchronization error" and "API HTTP 404" errors.

I use satstack_0.19.0_Windows-amd64 and Ledger Live 2.66.0. Do they work together?

**Comment by adrienlacombe-ledger** [comment]
> Has there been any progress on this since last December?
> 
> Because I too get the "Synchronization error" and "API HTTP 404" errors.
> 
> I use satstack_0.19.0_Windows-amd64 and Ledger Live 2.66.0. Do they work together?

nope, see https://github.com/LedgerHQ/satstack#requirements, LLD 2.52.0 max

**Comment by adrienlacombe-ledger** [comment]
> Okay, I will try some older version then. Thank you.
> 
> But I really wish Ledger, instead of various shitcoinery shenanigans and making their "cold-storage" wallets hot, would focus on what is really important - Bitcoin and running one's full-node.

np

Not me you need to convince ;)

**Comment by MartinacPlus** [comment]
> > Okay, I will try some older version then. Thank you.
> > But I really wish Ledger, instead of various shitcoinery shenanigans and making their "cold-storage" wallets hot, would focus on what is really important - Bitcoin and running one's full-node.
> 
> np
> 
> Not me you need to convince ;)

My remark was not directed to you personally, but to Ledger as a company ;)

I installed the ledger-live-desktop-2.52.0-win-x64.exe, but had to delete the %APPDATA% Ledger folder (with the folder there, I got error messages).
But now the satstack is running the re-scanning from the genesis block... Last time it took 3 days... Can this somehow be skipped?

**Comment by adrienlacombe-ledger** [comment]
> > > Okay, I will try some older version then. Thank you.
> > > But I really wish Ledger, instead of various shitcoinery shenanigans and making their "cold-storage" wallets hot, would focus on what is really important - Bitcoin and running one's full-node.
> > 
> > 
> > np
> > Not me you need to convince ;)
> 
> My remark was not directed to you personally, but to Ledger as a company ;)
> 
> I installed the ledger-live-desktop-2.52.0-win-x64.exe, but had to delete the %APPDATA% Ledger folder (with the folder there, I got error messages). But now the satstack is running the re-scanning from the genesis block... Last time it took 3 days... Can this somehow be skipped?

Strange indeed, maybe satstack shutdown abruptly last time, not sure.

**Comment by ChimpRocket** [comment]
Hi,

I've just started using Ledger and have hit this '404' issue on day 1.  Do I still need to run an outdated version of Ledger-Live to workaround the issue?  I'm on the latest 2.77.2 at the moment.  

Running up to date v26.0 node, satstack sync completed successfully. Since completing the scan, I get 404 when trying to import my BTC accounts. 

Thanks,
JS

Edit: Update on this, I removed 2.77.2 and went with 2.52.0 to get satstack working ok.  2.52.0 comes with its own set of niggles and bugs though.  I hope there's a better solution coming - such as native support for node connectivity in the Ledger Live app.

**Comment by ROunofF** [comment]
I gave this problem a shot and found 3 urls that returns 404:

1. The LLD wants to hit /btc/block/:block
Fix was simple, just add the route and use the GetBlock: #134 

2. LLD wants to hit /address/:addresses/txs
I was able to reuse the code from GetAddresses, but the output structure was different. Possibly paginated (not supported): #135 

3. LLD wants to his /address/:addresses/txs/pending
Assuming this is a "transient" state (after X confirmations, it's probably not pending anymore). I'm not sure it's worth the effort to implement and I don't know what LLD expects... If someone has the output and is willing to implement this go ahead...
I made it return an empty response since this was my case for all address I tested... Patch : #136 

I feel this is a few steps in the right direction!
